src/client/public/scripts/components/base-element.ts

```diff
@@ -1,17 +1,44 @@
 import {render} from '../../../../../node_modules/lit-html/lib/lit-extended.js';
 import {TemplateResult} from '../../../../../node_modules/lit-html/lit-html.js';
 
+type Debouncer = (task: Function) => Promise<void>;
+
+function createDebouncer(): Debouncer {
+  let queued = false;
+  return async (task: Function) => {
+    if (queued) {
+      return;
+    }
+    queued = true;
+    await Promise.resolve();
+    queued = false;
+    task();
+  };
+}
+
 export abstract class BaseElement extends HTMLElement {
+  private _debouncer: Debouncer;
```

Comment:
With TypeScript you could even initialize this inline and avoid writing out the constructor.

Reply:
++, done.